modules/seqexec/server/src/main/scala/seqexec/server/InstrumentControllerSim.scala

```diff
@@ -37,90 +34,82 @@ class InstrumentControllerSim(name: String, useTimeout: Boolean) {
   private val ConfigurationDelay = Seconds(5)
 
   @tailrec
-  private def observeTic(stop: Boolean, abort: Boolean, pause: Boolean, remain: Int, timeout: Option[Int]): TrySeq[ObserveCommand.Result] =
+  private def observeTic(stop: Boolean, abort: Boolean, pause: Boolean, remain: Int, timeout: Option[Int]): F[ObserveCommand.Result] =
     if(remain < tic) {
       Log.info(s"Simulate $name observation completed")
-      TrySeq(ObserveCommand.Success)
-    } else if(stop) TrySeq(ObserveCommand.Stopped)
-      else if(abort) TrySeq(ObserveCommand.Aborted)
+      (ObserveCommand.Success: ObserveCommand.Result).pure[F]
```

Comment:
You can sometimes get away with `ObserveCommand.Success.pure[F].widen` in cases like this. It kind of depends on what else is in the code path. You can certainly say `.pure[F].widen[ObserveCommand.Result]` but that's not much better. This will be better in Scala 3.